syntax = "proto3";

message MemoryInfo {
  int64 total = 1;
  int64 free = 2;
  int64 used = 3;
  int32 percentage = 4;
}

message Temperature {
  oneof temp_reading {
    int32 Celsius = 1;
    int32 Fahrenheit = 2;
  }
}

message GPUBasicInfo {
  string name = 1;
  int32 index = 2;
  string serial = 3;
  string uuid = 4;
}

message CommonMatrixInfo {
  Temperature temperature = 1;
  MemoryInfo memory_info = 2;
  int32 usage = 3;
}

message NVGPU {
  GPUBasicInfo info = 1;
  CommonMatrixInfo matrix = 2;
  int32 cuda_version = 3;
  string cuda_capacity = 4;
}


message CPU {

}